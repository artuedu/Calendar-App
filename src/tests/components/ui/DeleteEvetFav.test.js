import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { eventStartDelete } from '../../../actions/events';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('../../../actions/events', () => ({
    eventStartDelete: jest.fn()
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};
let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <DeleteEventFab />
    </Provider>
);

describe('Pruebas en DeleteEventFav', () => {
    
    test('Debe de mostrarse correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de llamar el eventStartDelete al hacer click', () => {
       
        wrapper.find('button').prop('onClick')();

        expect(eventStartDelete).toHaveBeenCalled();

    });
    
    
});
