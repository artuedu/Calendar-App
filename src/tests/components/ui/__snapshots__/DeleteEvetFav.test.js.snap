// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en DeleteEventFav Debe de mostrarse correctamente 1`] = `ReactWrapper {}`;
